resources/aws_ec2.go

```diff
@@ -486,6 +491,74 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsServer := obj.snsServer()
+	// shutdown the sns endpoint when we're done
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.TODO(), SnsServerShutdownTimeout*time.Second)
+		defer cancel()
+		if err := snsServer.Shutdown(ctx); err != nil {
+			if err != context.Canceled {
+				log.Printf("%s: error stopping sns endpoint: %s", obj, err)
+				return
+			}
+			log.Printf("%s: sns server shutdown cancelled", obj)
+		}
+	}()
+	obj.wg.Add(1)
+	// start the endpoint
+	go func() {
+		defer obj.wg.Done()
+		defer close(obj.awsChan)
+		if err := snsServer.ListenAndServe(); err != nil {
+			// when we shut down
```

Comment:
you forgot something important here (within the next 5 lines), what is it? :)

Reply:
return!